trace-forward/src/Trace/Forward/Configuration.hs

```diff
@@ -0,0 +1,48 @@
+module Trace.Forward.Configuration
+  ( AcceptorConfiguration (..)
+  , ForwarderConfiguration (..)
+  , HowToConnect (..)
+  , Host
+  , Port
+  ) where
+
+import           Control.Tracer (Tracer)
+import           Data.IORef (IORef)
+import           Data.Text (Text)
+import           Data.Word (Word16)
+import           Ouroboros.Network.Driver (TraceSendRecv)
+
+import           Trace.Forward.Protocol.Type
+
+type Host = Text
+type Port = Word16
+
+-- | Specifies how to connect to the peer.
+data HowToConnect
+  = LocalPipe    !FilePath    -- ^ Local pipe (UNIX or Windows).
+  | RemoteSocket !Host !Port  -- ^ Remote socket (host and port).
+  deriving Show
+
+-- | Acceptor configuration, parameterized by trace item's type.
+data AcceptorConfiguration lo = AcceptorConfiguration
+  { -- | The tracer that will be used by the acceptor in its network layer.
+    acceptorTracer    :: !(Tracer IO (TraceSendRecv (TraceForward lo)))
+    -- | The endpoint that will be used to listen to the forwarder.
+  , forwarderEndpoint :: !HowToConnect
+    -- | The request specifies how many 'TraceObject's will be requested.
+  , whatToRequest     :: !NumberOfTraceObjects
+    -- | 'IORef' that can be used as a brake: if an external thread sets
+    --   it to 'True', the acceptor will send 'MsgDone' message to the
+    --   forwarder and their session will be closed.
```

Comment:
What happens if the client is in a wrong state which does not allow to send `MsgDone`.  Will it timeout? Will the owning thread (that set's the `IORef`) will be notified (does it need to?).

Reply:
> What happens if the client is in a wrong state which does not allow to send `MsgDone`.

Yes, the client won't send `MsgDone` **_immediately_** after we set this `IORef` to `True`. But it will send it during the next request.

Reply:
Will this external thread block until the protocol exists? Using `IORef` suggests that it does not, will it then terminate the protocol abruptly without giving it a chance to send `MsgDone`?

Reply:
> Will this external thread block until the protocol exists?

No, this external thread just sets `IORef` to `True`, nothing more. And then, during the next attempt to send request, the library checks this `IORef` and handles it.

Reply:
I missed your reply.  I was asking about the thread that tries to shut down the acceptor not the acceptor it self.  How will it wait for it to finish?

Reply:
> I was asking about the thread that tries to shut down the acceptor not the acceptor it self. How will it wait for it to finish?

It doesn't wait for the finish of the acceptor. This thread only sets `IORef` to `True`, it's like he said to the acceptor: "Hey, you should stop, please send `MsgDone` to the forwarder as soon as you can".

Reply:
Does it mean it will leak a thread (the protocol handlers) if the acceptor will not finish quickly?

Reply:
Yes, theoretically it's possible that, if the handler (https://github.com/input-output-hk/cardano-node/blob/9f82616efe1f2836e7435f4b3c0f4d48d28c83d0/trace-forward/src/Trace/Forward/Network/Acceptor.hs#L164) takes 2 hours, the `IORef` will be checked (and `MsgDone` will be sent) only after 2 hours. :-)

Reply:
I see, so I'd suggest when you set it, you monitor the thread that runs the protocol handlers and kill it after let say 30 seconds (or maybe quicker) if it did not return by it self.  You can use `Async` in combination with `registerDelay` to achieve this.

Reply:
Done.